helix-core/src/auto_pairs.rs

```diff
@@ -282,20 +274,20 @@ fn handle_open(doc: &Rope, selection: &Selection, pair: &Pair) -> Transaction {
 
         let change = match next_char {
             Some(_) if !pair.should_close(doc, start_range) => {
-                len_inserted = pair.open.len_utf8();
+                len_inserted = 1;
```

Comment:
Shouldn't this count graphemes? The pairs could technically be multi chars (although the code doesn't properly handle this yet)

Reply:
No, this is being used to compute the selection, which uses char indices. Although yes, this will have to be reworked for multi-char auto pairs.

Reply:
No, as it turns out, [`char::len_utf8`](https://doc.rust-lang.org/std/primitive.char.html#method.len_utf8) returns the number of bytes of the `char`. But one `char` is always just one `char`, so it just so happens to be correct for ASCII, but goes too far for high Unicode characters.

Reply:
@archseer does this make sense?

Reply:
Ah, I think that makes a lot of sense. I was curious about the literal `1`s as well. This might be worth a comment somewhere? I'm not sure exactly where I would put it